### ISSUE TITLE
Invalid request headers due to machine name

### ISSUE THREAD

**Issue by StevenJohnston** [splitio/javascript-client#847]
On macOS my default machine name was `Steven’s-MacBook-Pro` This was causing split.io to emit `SDK_READY_TIMED_OUT`. I assume due to the `’` in the name.

What did **not** work was: 
```
const config: INodeSettings = {
      ...
      sync: {
        requestOptions: {
          getHeaderOverrides: (context) => {
            context.headers.SplitSDKMachineName = 'StevensMacBookPro';
            return context.headers;
          },
        },
      },
      ...
}
``` 

What **did** work was:
* changing my machines name via `sudo scutil --set HostName "stevensmacbookpro"`

**Comment by IngaAlma** [comment]
Hi Steven,
In the example above, you've entered the machine name without ': 
context.headers.SplitSDKMachineName = 'StevensMacBookPro';
Does this fail?

I was also able to run this configuration with no issues:
requestOptions: {
            getHeaderOverrides: (context) => {
                // Add a custom header
                context.headers.SplitSDKMachineName = "Steven's-Mac'Book-Pro";

                // Return modified headers
                return context.headers;
            }
        }, 

Best,
Inga

**Comment by StevenJohnston** [comment]
> ’
In my example getHeaderOverrides was not modifying my SplitSDKMachineName header. 

I think our quotes are differing `Steven’s-MacBook-Pro` `’` as  U+2019  and `Steven's-Mac'Book-Pro` `'` as U+0027.

**Comment by EmilianoSanchez** [comment]
Hi @StevenJohnston , 

Thanks for sharing your issue. I’ve checked on my side and confirmed that this is a bug in the SDK. It should eliminate or sanitize characters not ISO-8859-1, like `’`, from header values because the Node.js http/https API and Web fetch API throw an exception.

We’ll address this in the next release, but unfortunately, we don’t have an ETA yet.

Additionally, the `SplitSDKMachineName` header is one of the SDK’s special headers that cannot be overwritten using the `getHeaderOverrides` configuration option. For now, the only workaround I see is to update your hostname as you suggested.

I will keep the issue open, and close it when we fix the problem.

**Comment by EmilianoSanchez** [comment]
Actually, @StevenJohnston , there is another workaround: you can set the `core.IPAddressesEnabled` configuration option to `false`, so that the `SplitSDKMachineIP` and `SplitSDKMachineName` are not sent.

Example:

```
const config: INodeSettings = {
      core: {
        ...
        IPAddressesEnabled: false
      },
      ...
}
```

**Comment by EmilianoSanchez** [comment]
Hi @StevenJohnston ,

I’m closing this issue for now, as we’ve just released JavaScript SDK v11.0.3, which includes a fix. Try this new version on your side.

Thank you
